### PR TITLE
Configure leader election with completely separate k8s client

### PR DESCRIPTION
**Pull request by rramkumar1**
Due to some scaling issues, we need to configure leader election with a separate k8s client then what is used for other API server operations (watch, events, etc). 

This puts ingress-gce in line w/ how scheduler and controller-manager implement leader election.

### CODE REVIEW

File: cmd/glbc/main.go [hunk 1]
```diff
@@ -83,6 +83,12 @@ func main() {
 		glog.Fatalf("Failed to create kubernetes client: %v", err)
 	}
 
+	// Due to scaling issues, the leader election must be configured with a separate k8s client.
+	leaderElectKubeClient, err := kubernetes.NewForConfig(kubeConfig)
```
[the remark this answers is not in the document]
Comment (rramkumar1) [reply cmd/glbc/main.go]: Done

**Comment by bowei** [comment]
@wojtek-t -- maybe the leader election code should create the client internally for the default use case? This seems like an easy thing to mess up.

**Comment by wojtek-t** [comment]
> @wojtek-t -- maybe the leader election code should create the client internally for the default use case? This seems like an easy thing to mess up.

I'm not the one who created this library and I'm not really maintaining it. But sure - that makes a lot of sense to me. Making kubeconfig a parameter of constructor for leader election makes a lot of sense.

**Pull request merged**